### ISSUE TITLE
rhdf5 broken on 21.11-pre

### ISSUE THREAD

**Issue by ivan-tkatchev** [NixOS/nixpkgs#125101]
**Describe the bug**
The rhdf5 library does not work on latest release.

**To Reproduce**
```
nix-shell -I 'nixpkgs=https://github.com/NixOS/nixpkgs/archive/refs/tags/21.11-pre.tar.gz' -p 'rWrapper.override { packages = [ rPackages.rhdf5 ]; }'
```

**Expected behavior**
A shell with R and rhdf5.

**Additional context**
Instead I get:
```
these derivations will be built:
  /nix/store/62rbp1iwk3qyzrwa7xffc0hhvlf0xry2-rhdf5_2.32.4.tar.gz.drv
  /nix/store/vsagg33zz18jmhsshqqrl5k499g2iqph-r-Rhdf5lib-1.10.1.drv
  /nix/store/1r21pqhdrnnxm7iic43hnn5lmaqjsi7w-r-rhdf5-2.32.4.drv
  /nix/store/nnh13xysfxpll6cd9c21d2drahways9x-R-4.0.4-wrapper.drv
these paths will be fetched (1.46 MiB download, 5.87 MiB unpacked):
  /nix/store/3xyij4rjz3z4yzj4incazjwiirkaz4f4-nghttp2-1.43.0-bin
  /nix/store/4x861aprr5ljaa2prcvhyxcrara8rmyb-openssl-1.1.1k-dev
  /nix/store/cdib8alblidlnj6vh27l1mi9slksm7vr-openssl-1.1.1k-bin
  /nix/store/fg560vvb30i52v4gm93ah513c6swqna7-curl-7.76.1-bin
  /nix/store/gvb3z5nqzw8m9s4kmifxq8jh5dqnnxjw-nghttp2-1.43.0-dev
  /nix/store/jm6zis2b51h8immqv7s32xwzln8xx2iv-curl-7.76.1-dev
  /nix/store/k09ayvnk26x86v3m1q4spi458sbsvyzz-libssh2-1.9.0
  /nix/store/szawrh0qpa2mg1w9krlrh2mkzw271ncx-curl-7.76.1
  /nix/store/w140ykfr9imm7nflzqwz4by7yr1dh3mw-libssh2-1.9.0-dev
building '/nix/store/vsagg33zz18jmhsshqqrl5k499g2iqph-r-Rhdf5lib-1.10.1.drv'...
unpacking sources
unpacking source archive /nix/store/4jvc6sjfg9fv5xmxmnp6fh830lgalhyi-Rhdf5lib_1.10.1.tar.gz
source root is Rhdf5lib
setting SOURCE_DATE_EPOCH to timestamp 1594346363 of file Rhdf5lib/configure
patching sources
applying patch /nix/store/fqsil1vhdy3m7w983pfcb54sd4cs5kb0-Rhdf5lib.patch
patching file configure
Hunk #1 FAILED at 2880.
1 out of 1 hunk FAILED -- saving rejects to file configure.rej
builder for '/nix/store/vsagg33zz18jmhsshqqrl5k499g2iqph-r-Rhdf5lib-1.10.1.drv' failed with exit code 1
cannot build derivation '/nix/store/1r21pqhdrnnxm7iic43hnn5lmaqjsi7w-r-rhdf5-2.32.4.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/nnh13xysfxpll6cd9c21d2drahways9x-R-4.0.4-wrapper.drv': 1 dependencies couldn't be built
error: build of '/nix/store/nnh13xysfxpll6cd9c21d2drahways9x-R-4.0.4-wrapper.drv' failed
```

**Notify maintainers**
@iimog 

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.15.0-135-generic, Ubuntu, 18.04.3 LTS (Bionic Beaver)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(ivan): `""`
 - channels(root): `"nixpkgs-21.03pre245659.30e98a4fd67"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by ivan-tkatchev** [comment]
"6.topic: rust" should be removed, this is an R package, not Rust.

**Comment by FliegendeWurst** [comment]
This was fixed in #125556
